Objects/genobject.c

```diff
@@ -471,14 +471,11 @@ _gen_throw(PyGenObject *gen, int close_on_genexit,
                 return gen_send_ex(gen, Py_None, 1, 0);
             goto throw_here;
         }
+        PyThreadState *tstate = _PyThreadState_GET();
         if (PyGen_CheckExact(yf) || PyCoro_CheckExact(yf)) {
             /* `yf` is a generator or a coroutine. */
-            PyThreadState *tstate = _PyThreadState_GET();
-            /* Since we are fast-tracking things by skipping the eval loop,
-               we need to update the current frame so the stack trace
-               will be reported correctly to the user. */
-            /* XXX We should probably be updating the current frame
-               somewhere in ceval.c. */
```

Comment:
Why remove this comment?

Reply:
I don't feel that strongly about it, but I would suggest the comment I've replaced it with more precisely describes why linking the frame is needed. I found the existing comment a bit confusing because in my view we're not "skipping" the eval loop - as implemented there's no way we can go through it, and this is just how throwing into generator chains works.

As for the `XXX` part, I think that's a bit debatable and I doubt anyone has plans to fix it. It's just a TODO lying around in the code that will never get addressed. If it does, it probably won't be because of this comment.

FWIW, I appreciate I'm not well known in the OSS community but I have quietly been working away on Python generator implementations for a while, first as part of [Skybison](https://github.com/facebookarchive/skybison) and these days as part of [Cinder](https://github.com/facebookincubator/cinder)'s JIT. So I do have a bit of context.

Reply:
OK, good enough for the first part. I'd rather keep the `XXX` though. Yes, it's not that it's useful as a TODO, but that's not really what they're helpful for. If there was a bug someday that resulted from not implementing the `XXX`, then it would be helpful for maintainers as they would get a pretty good idea of where to start with the fix.